src/ui/lib/zcl_abapgit_html_table.clas.testclasses.abap

```diff
@@ -226,4 +227,71 @@
 
   ENDMETHOD.
 
+  METHOD with_groups.
+
+    DATA lo_tab TYPE REF TO zcl_abapgit_html_table.
+    DATA lv_html_act TYPE string.
+    DATA li_html_exp TYPE REF TO zif_abapgit_html.
+    DATA ls_sort TYPE zif_abapgit_html_table=>ty_sorting_state.
+
+    lo_tab = zcl_abapgit_html_table=>create( me
+      )->define_column_group(
+      )->define_column(
+        iv_column_id    = 'col1'
+        iv_column_title = 'Col 1'
+      )->define_column_group(
+        iv_group_title = 'Group'
+      )->define_column(
+        iv_column_id    = 'col2'
+        iv_column_title = 'Col 2'
+      )->define_column(
+        iv_column_id    = 'col3'
+        iv_column_title = 'Col 3'
+        iv_sortable     = abap_false ).
+
+    ls_sort-column_id = 'col1'.
+
+    lv_html_act = lo_tab->render(
+      is_sorting_state = ls_sort
+      iv_id        = 'with-sort'
+      it_data      = test_data_set( ) )->render( ).
+
+    CREATE OBJECT li_html_exp TYPE zcl_abapgit_html.
+
+    li_html_exp->add(
+      '<table id="with-sort">' )->add(
+      '<thead>' )->add(
+      '<tr>' )->add(
+      '<th></th>' )->add(
+      '<th colspan="2">Group</th>' )->add(
+      '</tr>' )->add(
+
+      '<tr>' )->add(
```

Comment:
Although it adds readability ...

<img width="530" alt="image" src="https://github.com/abapGit/abapGit/assets/15635498/031237a2-880f-4f65-abb3-c82249efb9d5">

Reply:
reads much better with `add` in the front

```abap
li_html->add( '<table>'
  )->add( '<thead>'
  )->add( '<tr>'
  )->add( '<th>hello</th>'
  )->add( '<th>world</th>'
  )->add( '</tr>' ).

```

Reply:
Yes, this can be arranged actually, but another problem is max length of 120 symbols and that orphan `add` in the middle  does not fit.

Actually, I think this linter rule is mostly good ... but easy (in-code) exceptions would be so great ...

Reply:
Ah, you mean to break chaining in the middle. I didn't think about it, a good idea actually, thanks

Reply:
I wonder if abap will ever support multiline strings like in js ...

Reply:
Nah, multi line strings are confusing because you don't know if the line break is part of the string of not.